### ISSUE TITLE
Vaadin Services JS client

### ISSUE THREAD

**Issue by vlukashov** [vaadin/vaadin-services#2]
As a developer I want to access Vaadin Service methods available on my backend using a convenient JavaScript API, so that I do not have to put effort into issuing low-level HTTP requests to the `/:service/:method` endpoint and handling auth, serialization, possible failures, etc

DoD:
 - there is a `VaadinClient` JS class with a `call(service, method, ...parameters)` method
 - the `call` method returns a promise that gets resolved with the return value / rejected with the error when the backend call is completed / failed
 - the following snippet works:
   ```javascript
   import {VaadinClient} from '@vaadin/services-core';

   const client = new VaadinClient({ endpoint: 'http://127.0.0.1:8080/services' });
   const result = await client.call('HelloService', 'hello', 'world');
   console.log(result); // prints 'Hello, world!'
   ```

The [common DoD](https://github.com/vaadin/vaadin-services/wiki/Definition-of-Done) applies.

**Comment by Legioth** [comment]
`...parameters` implies that parameters are passed by order instead of by name. In previous discussions, we have been leaning towards passing by name. Is this an accidental omission, or do we have some newly found reason for preferring to pass by index?

I'd say that points if favour of passing by index are:
* It's more compact since the parameter names do not need to be sent over the wire
* It's easier to receive values on the server because we don't need to work around the fact that Java reflection doesn't give access to parameter names.

Benefits of passing by name are:
* It's more REST-like
* It's less fragile if the client and the server are for some reason using different versions of the method signature.

**Comment by vlukashov** [comment]
Indeed, we've agreed on using an single object dictionary for call parameters rather than passing them as a var args array.

**Comment by vlukashov** [comment]
I would be opposed to putting such defaults into the npm package that's common for all users because it subjectively seems to be too arbitrary for a global default. There are plans to generate project-specific default client configuration, and I see that default client config as a better place for the default endpoint.

**Comment by Legioth** [comment]
In what way would it be better to have a global default that doesn't work at all without explicit configuration? If there's a simple configuration approach that could work in maybe 90% of all situations, then why shouldn't that approach be used as the default?

**Comment by vlukashov** [comment]
Outside of scope of this task:
 - endpoint URL should be configurable on the backend as well
 - CORS should be supported (eventually)

Other comments:
 - full ent-to-end integration test may be done as a separate task if takes too long
 - using the fetch API is OK as along as there is a polyfill for IE11 and Safari 9
